gunrock/app/mf/mf_app.cu

```diff
@@ -37,137 +41,142 @@ cudaError_t UseParameters(util::Parameters &parameters)
     GUARD_CU(UseParameters_problem(parameters));
     GUARD_CU(UseParameters_enactor(parameters));
 
-    // TODO: add app specific parameters, e.g.:
-     GUARD_CU(parameters.Use<std::string>(
-        "src",
-        util::REQUIRED_ARGUMENT | util::MULTI_VALUE | util::OPTIONAL_PARAMETER,
-        "0",
-        "<Vertex-ID|random|largestdegree> The source vertices\n"
-        "\tIf random, randomly select non-zero degree vertices;\n"
-        "\tIf largestdegree, select vertices with largest degrees",
-        __FILE__, __LINE__));
-
-     GUARD_CU(parameters.Use<int>(
-        "src-seed",
-        util::REQUIRED_ARGUMENT | util::SINGLE_VALUE | util::OPTIONAL_PARAMETER,
-        util::PreDefinedValues<int>::InvalidValue,
-        "seed to generate random sources",
-        __FILE__, __LINE__));
-
+    GUARD_CU(parameters.Use<uint32_t>(
+	"source",
+	util::REQUIRED_ARGUMENT,
+	util::PreDefinedValues<uint32_t>::InvalidValue,
+	"<Vertex-ID|random|largestdegree> The source vertex\n"
+	"\tIf random, randomly select non-zero degree vertex;\n"
+	"\tIf largestdegree, select vertex with largest degree",
+	__FILE__, __LINE__));
+
+    GUARD_CU(parameters.Use<uint32_t>(
+	"sink",
+	util::REQUIRED_ARGUMENT,
+	util::PreDefinedValues<uint32_t>::InvalidValue,
+	"<Vertex-ID|random|largestdegree> The source vertex\n"
+	"\tIf random, randomly select non-zero degree vertex;\n"
```

Comment:
This will fail, I think. You are expecting a `uint32_t`, but setting the `--sink=random` would result in an error due to invalid type, similar case for `largestdegree`. Unless I am understanding this incorrectly?

Reply:
True, only Vertex-ID is OK